pkg/keys/ed25519.go

```diff
@@ -150,12 +171,18 @@ func (e *ed25519Signer) UnmarshalPrivateKey(key *data.PrivateKey) error {
 	return nil
 }
 
-func (e *ed25519Signer) PublicData() *data.PublicKey {
-	keyValBytes, _ := json.Marshal(ed25519Verifier{PublicKey: []byte(e.PrivateKey.Public().(ed25519.PublicKey))})
+func (e *ed25519Signer) PublicData() (*data.PublicKey, error) {
+	keyValBytes, err := json.Marshal(ed25519Verifier{
```

Comment:
Similarly, I'd revert this

Reply:
no need to remarshall or the breaking change

Reply:
Actually, nevermind—you can leave this with the new panic instead of ignoring the error.